sdk/core/core/inc/az_span.h

```diff
@@ -102,15 +102,13 @@ AZ_NODISCARD AZ_INLINE az_result az_error_unexpected_char(az_result_byte const c
   return az_failed(c) ? c : AZ_ERROR_PARSER_UNEXPECTED_CHAR;
 }
 
-AZ_NODISCARD az_result az_span_get_uint64(az_span const self, uint64_t * const out);
-
-#define AZ_ARRAY_SIZE(ARRAY) (sizeof(ARRAY) / sizeof(*ARRAY))
+AZ_NODISCARD az_result az_span_to_uint64(az_span const self, uint64_t * const out);
 
 #define AZ_SPAN_FROM_ARRAY(ARRAY) \
-  { .begin = ARRAY, .size = AZ_ARRAY_SIZE(ARRAY) }
+  { .begin = ARRAY, .size = (sizeof(ARRAY) / sizeof(*ARRAY)) }
 
-AZ_NODISCARD AZ_INLINE az_span az_span_from_one(uint8_t const * ptr) {
-  return (az_span){ .begin = ptr, .size = 1 };
+AZ_NODISCARD AZ_INLINE az_span az_span_from_buffer_range(uint8_t const * ptr, size_t const size) {
```

Comment:
`az_span_from_one` should be used when we have a `uint8_t` variable. In this case, `size` doesn't make sense. If you need a function for sub span, use `az_span_drop`, `az_span_take`, `az_span_sub`.